src/cpp/session/resources/terminal/hooks

```diff
@@ -59,9 +59,9 @@ if [ -n "${_RS_CONDA_PREFIX}" ]; then
 
 fi
 
-if [ -f "${RETICULATE_PYTHON}" ]; then
+if [ -f "${RETICULATE_PYTHON_FALLBACK}" ]; then
```

Comment:
So what happens is, if RETICULATE_PYTHON is set, then infer reticulate Python will return the value of RETICULATE_PYTHON, and RETICULATE_PYTHON_FALLBACK will automatically get set to the value of RETICULATE_PYTHON. Therefore we should always be able to use the RETICULATE_PYTHON_FALLBACK variable, instead of checking both, because FALLBACK will point to RETICULATE_PYTHON whenever that's set.

Reply:
Actually thinking about this again, this is true if the user is in a project and has selected that option, but if the user is not in a project, or has not checked that option, then in that case I believe it is possible for RETICULATE_PYTHON to be set and RETICULATE_PYTHON_FALLBACK to be unset. I'll handle that

Reply:
Okay, I double checked this, because you raised a good question. What I said in my first comment is correct. 

Let's say a user has set `RETICULATE_PYTHON` in their `~/.Renviron`, and let's say that they are not in a project.

.Renviron
<img width="600" alt="image" src="https://user-images.githubusercontent.com/7817881/155161396-e7860140-17c1-49b2-ae91-c08f4228bcfc.png">

Environment variables in the console
<img width="479" alt="image" src="https://user-images.githubusercontent.com/7817881/155161469-9795b5d4-09c0-4116-89c5-c5b812d7fc98.png">
Note that `.rs.inferReticulatePython()` correctly returns the value set by `RETICULATE_PYTHON`

In the RStudio Terminal
<img width="479" alt="image" src="https://user-images.githubusercontent.com/7817881/155161660-b0919859-b8fb-4535-9b68-e2cb07b23fbf.png">
Note that, because my `RETICULATE_PYTHON` points to a virtual env python, in the terminal, that virtual env is activated (even though I did not call activate), `which python` points to the correct python, and both `RETICULATE_PYTHON` and `RETICULATE_PYTHON_FALLBACK` system environment variables are set in the terminal (even though only `RETICULATE_PYTHON`) is set in the R environment.

Therefore, I feel comfortable that the answer to your question is yes, this is going to do the right thing.